agent/eni/watcher/watcher_windows_test.go

```diff
@@ -0,0 +1,700 @@
+func TestEventHandlerUnmanagedENI(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	ctx := context.TODO()
+	eventChannel := make(chan statechange.Event)
+
+	mockiphelper := mock_iphelperwrapper.NewMockInterfaceMonitor(mockCtrl)
+	mocknetwrapper := mock_gonetwrapper.NewMockNetWrapper(mockCtrl)
+	mockStateManager := mock_dockerstate.NewMockTaskEngineState(mockCtrl)
+
+	mac1, _ := net.ParseMAC(macAddress1)
+	gomock.InOrder(
+		mockiphelper.EXPECT().Start(gomock.Any()).Return(nil),
+		mocknetwrapper.EXPECT().FindInterfaceByIndex(interfaceIndex1).Return(&net.Interface{
+			Index:        interfaceIndex1,
+			Name:         interfacename1,
+			HardwareAddr: mac1,
+		}, nil),
+		mockStateManager.EXPECT().ENIByMac(macAddress1).
+			Return(nil, false).AnyTimes(),
+	)
+
+	watcher, _ := newTestWatcher(ctx, primaryMAC, mockStateManager, eventChannel, mockiphelper)
+	netutils := networkutils.New()
+	netutils.SetNetWrapper(mocknetwrapper)
+	watcher.SetNetworkUtils(netutils)
+
+	go watcher.eventHandler()
+	watcher.notifications <- interfaceIndex1
+
+	time.Sleep(cancelContextDelay)
+	var wait sync.WaitGroup
+	wait.Add(1)
+
+	mockiphelper.EXPECT().Close().DoAndReturn(
+		func() error {
+			wait.Done()
+			return nil
+		})
+	go watcher.Stop()
+	wait.Wait()
+}
+
+// Test for eventHandle method. In this test case, we receive a notification about an ENI whose timer has already expired.
+// Therefore, that ENI will be removed from State
+func TestEventHandlerExpiredENI(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	ctx := context.TODO()
+	eventChannel := make(chan statechange.Event)
+
+	mockiphelper := mock_iphelperwrapper.NewMockInterfaceMonitor(mockCtrl)
+	mocknetwrapper := mock_gonetwrapper.NewMockNetWrapper(mockCtrl)
+	mockStateManager := mock_dockerstate.NewMockTaskEngineState(mockCtrl)
+
+	mac1, _ := net.ParseMAC(macAddress1)
+	gomock.InOrder(
+		mockiphelper.EXPECT().Start(gomock.Any()).Return(nil),
+		mocknetwrapper.EXPECT().FindInterfaceByIndex(interfaceIndex1).Return(&net.Interface{
+			Index:        interfaceIndex1,
+			Name:         interfacename1,
+			HardwareAddr: mac1,
+		}, nil),
+		mockStateManager.EXPECT().ENIByMac(macAddress1).
+			Return(&apieni.ENIAttachment{
+				MACAddress: macAddress1,
+				ExpiresAt:  time.Now().Add(expirationTimeSubtraction),
+			}, true),
+		mockStateManager.EXPECT().RemoveENIAttachment(macAddress1),
+	)
+
+	watcher, _ := newTestWatcher(ctx, primaryMAC, mockStateManager, eventChannel, mockiphelper)
+	netutils := networkutils.New()
+	netutils.SetNetWrapper(mocknetwrapper)
+	watcher.SetNetworkUtils(netutils)
+
+	go watcher.eventHandler()
+	watcher.notifications <- interfaceIndex1
+
+	time.Sleep(cancelContextDelay)
+	var wait sync.WaitGroup
+	wait.Add(1)
+
+	mockiphelper.EXPECT().Close().DoAndReturn(
+		func() error {
+			wait.Done()
+			return nil
+		})
+	go watcher.Stop()
+	wait.Wait()
+}
+
+// These are tests for common functionality of Linux and Windows
```

Comment:
Can you make the common tests to a different file ?

Reply:
Refactored these tests in a common file and deleted the same from Linux and Windows test file.